[CI] XPackRestIT test {p0=esql/10_basic/basic with documents_found} failing

**Build Scans:**
- [elasticsearch-periodic #9183 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/q33ohis5kqf46)
- [elasticsearch-periodic-platform-support #9189 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/hnb6segaqw3ji)
- [elasticsearch-periodic-platform-support #9175 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ntfm7cuf4ozjo)
- [elasticsearch-periodic-platform-support #9154 / windows-2025_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/paoocdxheewic)
- [elasticsearch-periodic #9105 / openjdk17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/5nl2czpg4lrp6)
- [elasticsearch-intake #24877 / part3](https://gradle-enterprise.elastic.co/s/trtubludn5dic)
- [elasticsearch-periodic #9077 / openjdk23_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/f7iow3d572qhg)
- [elasticsearch-periodic-platform-support #9077 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/e3cyrgw6pj5za)
- [elasticsearch-periodic-platform-support #9062 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/vdeibi3kgumgm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=esql/10_basic/basic with documents_found}" -Dtests.seed=99AD545FAF01E145 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=vec-Latn-IT -Dtests.timezone=America/Ojinaga -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Desql%2F10_basic%2Fbasic%20with%20documents_found%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/10_basic:200]: 
Expected: <10>
     but: was <9>
```

**Issue Reasons:**
- [8.19] 9 failures in test test {p0=esql/10_basic/basic with documents_found} (1.2% fail rate in 722 executions)
- [8.19] 2 failures in step almalinux-8_platform-support-unix (11.8% fail rate in 17 executions)
- [8.19] 3 failures in pipeline elasticsearch-periodic (15.0% fail rate in 20 executions)
- [8.19] 5 failures in pipeline elasticsearch-periodic-platform-support (25.0% fail rate in 20 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -503,9 +503,6 @@ tests:
 - class: org.elasticsearch.compute.aggregation.TopIntAggregatorFunctionTests
   method: testManyInitialManyPartialFinalRunnerThrowing
   issue: https://github.com/elastic/elasticsearch/issues/130145
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=esql/10_basic/basic with documents_found}
-  issue: https://github.com/elastic/elasticsearch/issues/130256
 - class: org.elasticsearch.xpack.searchablesnapshots.cache.shared.NodesCachesStatsIntegTests
   method: testNodesCachesStats
   issue: https://github.com/elastic/elasticsearch/issues/129863

